extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/ContractAgreementDataFetcher.java

```diff
@@ -76,10 +79,44 @@ public List<ContractAgreementData> getContractAgreements(DSLContext dsl) {
             .toList();
     }
 
-    private @NotNull Map<String, SovityContractTerminationRecord> fetchTerminations(DSLContext dsl, List<ContractAgreement> agreements) {
+    @NotNull
+    public ContractAgreementData getContractAgreement(DSLContext dsl, String contractAgreementId) {
 
-        var agreementIds = agreements.stream().map(ContractAgreement::getId).toList();
+        val agreement = getAllContractAgreement(contractAgreementId);
+
+        val negotiationQuery = QuerySpec.max();
+        val negotiation = contractNegotiationStore.queryNegotiations(negotiationQuery)
+            .filter(it -> it.getContractAgreement().getId().equals(contractAgreementId))
+            .findFirst()
+            .orElseThrow(
+                () -> new IllegalStateException("Can't find any negotiation for contract agreement id %s".formatted(contractAgreementId)));
+
+        val transfers = getAllTransferProcesses().stream().collect(groupingBy(it -> it.getDataRequest().getContractId()));
+
+        val terminations = fetchTerminations(dsl, agreement.getId());
+
+        val maybeAsset = assetIndex.findById(agreement.getAssetId());
+        val isConsumer = negotiation.getType() == ContractNegotiation.Type.CONSUMER;
+        val asset = (maybeAsset == null || isConsumer) ? dummyAsset(agreement.getAssetId()) : maybeAsset;
+
+        return new ContractAgreementData(
+            agreement,
+            negotiation,
+            asset,
+            transfers.getOrDefault(agreement.getId(), List.of()),
+            terminations.get(agreement.getId())
+        );
+    }
+
+    private ContractAgreement getAllContractAgreement(String id) {
```

Comment:
Sorry this method name is confusing as hell. What does the "all" mean here? As far as I see it returns simply a contract agreement. I mean, I know getContractAgreement would collide with the method above but this ain't it

Reply:
It means that I copy/pasted it and my brained didn't read the name and only deleted the `s` in the end to make it singular :-). Yes, also during the self review.
Renamed to `getContractAgreementById`